public/app/features/search/hooks/useManageDashboards.ts

```diff
@@ -42,14 +42,13 @@ export const useManageDashboards = (
     dispatch({ type: MOVE_ITEMS, payload: { dashboards: selectedDashboards, folder } });
   };
 
-  const canMove = useMemo(() => results.some((result) => result.items && result.items.some((item) => item.checked)), [
-    results,
-  ]);
+  const canMove = useMemo(() => results.some((result) => result.items.some((item) => item.checked)), [results]);
 
   const canDelete = useMemo(() => {
+    const somethingChecked = results.some((result) => result.checked || result.items.some((item) => item.checked));
```

Comment:
nit: I wonder if it would be worth to flatten everything and check for checked because using `.some` in another `.some` isn't super readable. I know I've been part of adding to this complexity in the past 🙈   I guess as long we have good tests for these cases its not super important. 🤔

Reply:
@hugohaggmark yeah agreed it's not super readable, but also didn't wanna flatten as that means we definitely have to go through the entire array. when using list view that means looping an array containing every dashboard that exists in your instance 😬  whereas with the `.some` it shortcircuits that loop as soon as it finds one. 

have moved the logic into a separate mini function instead. wdyt? 🤔

Reply:
You're right! Looks better as a separate function.